block-storage/src/main/scala/coop/rchain/blockstorage/dag/BlockDagStorage.scala

```diff
@@ -37,6 +42,8 @@ trait BlockDagRepresentation[F[_]] {
       startBlockNumber: Long,
       maybeEndBlockNumber: Option[Long]
   ): F[Vector[Vector[BlockHash]]]
+  // DAG representation has to have finalized block, or it does not make sense
+  def lastFinalizedBlock: BlockHash
```

Comment:
I think this can be a syntax method and calculated by just traversing the dag from latestBlock to the latest finalized block. 
It is fine to leave it here. Just mention my idea.

Reply:
We are storing `lastFinalizedBlock` in metadata, but what you describe is how it's calculated. DagRepresentation should have lastFinalizedBlock or it does not make sense because at least genesis or some block should be initial point of the DAG evolution.

Reply:
You understand me correctly.
What I meant ultimately is that, maybe we don't need to store lfb. Just an idea. Not a request.

Reply:
I think we need to store this only for merging purposes. Actually only two the most recent LFBs are needed. But might be it worth to store them all though, not that much data.